### PR TITLE
Fixed LoomDemo to allow argument passing and console on Windows to work.

### PR DESCRIPTION
**Pull request by SmilyOrg**
Included are also fixes for shutdown cleanup, better support for direct LoomDemo execution with console on Windows and removal of OpenAL message complaining about having to cleanup resources on shutdown.

### CODE REVIEW

File: loom/vendor/openal-soft/Alc/ALc.c [hunk 1]
```diff
@@ -2859,7 +2859,7 @@ static void ReleaseALC(void)
         do {
             num++;
         } while((dev=dev->next) != NULL);
-        ERR("%u device%s not closed\n", num, (num>1)?"s":"");
+        //ERR("%u device%s not closed\n", num, (num>1)?"s":"");
```
Comment (bengarney) [inline loom/vendor/openal-soft/Alc/ALc.c]: Note why we commented this out please!

File: application/windows/main.cpp [hunk 2]
```diff
@@ -84,22 +88,50 @@ int CALLBACK WinMain( HINSTANCE hInstance, HINSTANCE hPrevInstance, LPSTR lpCmdL
...
+            fromRuby = true;
+
             char *pEnd;  
 
             long int pid = strtol (argv[i + 1], &pEnd, 10);
 
             CCApplication::sharedApplication().setCLIRubyProcessId(pid);
-            
+
+            memmove(argv + i, argv + i + 2, (argc - i - 2)*sizeof(char*));
```
Comment (bengarney) [inline application/windows/main.cpp]: Are you sure about this memmove?

File: application/windows/main.cpp [hunk 3]
```diff
@@ -84,22 +88,50 @@ int CALLBACK WinMain( HINSTANCE hInstance, HINSTANCE hPrevInstance, LPSTR lpCmdL
...
+            argc -= 2;
+            i--;
             break;
         }
     }
 
+    LS::Process::consoleAttached = false;
+    if (!fromRuby && AttachConsole(ATTACH_PARENT_PROCESS))
+    {
+        HANDLE consoleHandleOut = GetStdHandle(STD_OUTPUT_HANDLE);
```
Comment (bengarney) [inline application/windows/main.cpp]: The money!

**Comment by bengarney** [comment]
Nitted!

File: application/windows/main.cpp (hunk 2 above)
Comment (SmilyOrg) [inline application/windows/main.cpp]: Pretty sure, tested it quite a bit.

File: application/windows/main.cpp (hunk 3 above)
Comment (SmilyOrg) [inline application/windows/main.cpp]: Yes!

**Comment by bengarney** [comment]
Thanks. This looks good. Can you be available to pull it down from firehose and test it when I land?

**Comment by SmilyOrg** [comment]
Ah, can we land tomorrow then? It's getting late.

**Comment by bengarney** [comment]
Fine with me!

**Comment by SmilyOrg** [comment]
Tested on firehose and it works!

**Pull request merged**